File: backend/src/AI.TestPlatform.Domain/Entities/Requirement.cs
namespace AI.TestPlatform.Domain.Entities;

/// <summary>需求状态：未开始 → 进行中 → 已完成</summary>
public enum RequirementStatus { NotStarted = 0, InProgress = 1, Completed = 2 }

/// <summary>
/// 需求（精简版）：测试覆盖的锚点。不做需求管理全流程（那不是测试平台的事），
/// 只回答两个问题：这个需求有没有用例在测？哪些需求还裸奔？
/// 外部需求系统（禅道/Jira）的正式对接预留 ExternalKey。
/// </summary>
public class Requirement
{
    public Guid Id { get; set; }
    public Guid ProjectId { get; set; }
    public Project Project { get; set; } = null!;

    /// <summary>需求标题（如"用户登录-密码错误提示"）</summary>
    public string Title { get; set; } = string.Empty;

    /// <summary>补充说明：验收要点、来源文档链接等</summary>
    public string? Description { get; set; }

    /// <summary>外部需求系统的标识（对接预留）</summary>
    public string? ExternalKey { get; set; }

    /// <summary>优先级：P0/P1/P2（自由文本，排序提示用）</summary>
    public string? Priority { get; set; }

    // ------------------------------ 进度字段
    /// <summary>计划开始时间</summary>
    public DateTime? PlanStartDate { get; set; }
    /// <summary>计划完成时间</summary>
    public DateTime? PlanEndDate { get; set; }
    /// <summary>实际开始时间</summary>
    public DateTime? ActualStartDate { get; set; }
    /// <summary>实际完成时间</summary>
    public DateTime? ActualEndDate { get; set; }
    /// <summary>状态：未开始 / 进行中 / 已完成</summary>
    public RequirementStatus Status { get; set; } = RequirementStatus.NotStarted;

    // ------------------------------ 审计字段（由 TestDbContext 统一盖章）
    /// <summary>创建人</summary>
    public Guid? CreatedById { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    /// <summary>最后修改人</summary>
    public Guid? UpdatedById { get; set; }
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>关联的用例（一个用例只挂一个需求；反向统计覆盖率）</summary>
    public List<TestCase> TestCases { get; set; } = new();

    /// <summary>关联的测试计划（反向：一个需求可以有多个计划在测）</summary>
    public List<TestPlan> TestPlans { get; set; } = new();
}

File: frontend/src/types/requirement.ts
/** 需求状态枚举（与后端 RequirementStatus 数值一致） */
export const RequirementStatus = {
  NotStarted: 0,
  InProgress: 1,
  Completed: 2,
} as const

export type RequirementStatusValue = (typeof RequirementStatus)[keyof typeof RequirementStatus]

export const REQUIREMENT_STATUS_LABELS: Record<number, string> = {
  [RequirementStatus.NotStarted]: '未开始',
  [RequirementStatus.InProgress]: '进行中',
  [RequirementStatus.Completed]: '已完成',
}

/** 需求（覆盖统计锚点，精简版） */
export interface Requirement {
  id: string
  projectId: string
  title: string
  description?: string | null
  externalKey?: string | null
  priority?: string | null
  createdAt: string
  /** 创建人显示名（M8 审计字段；历史行可能为空） */
  createdByName?: string | null
  // ------------------------------ 进度字段
  planStartDate?: string | null
  planEndDate?: string | null
  actualStartDate?: string | null
  actualEndDate?: string | null
  /** 状态：0 未开始 / 1 进行中 / 2 已完成 */
  status?: RequirementStatusValue
}

/** 列表项：带覆盖统计 + 关联计划数 */
export interface RequirementListItem extends Requirement {
  /** 关联的可见用例数 */
  caseCount: number
  /** 关联用例中最近一次执行已通过的数量 */
  passedCaseCount: number
  /** 关联的测试计划数量 */
  linkedPlanCount: number
}

export interface RequirementCoverage {
  projectId: string
  totalRequirements: number
  coveredRequirements: number
  uncoveredRequirements: number
  /** 覆盖率（百分数，1 位小数） */
  coverageRate: number
  uncoveredList: RequirementListItem[]
}

export interface RequirementPayload {
  title: string
  description?: string | null
  externalKey?: string | null
  priority?: string | null
  planStartDate?: string | null
  planEndDate?: string | null
  actualStartDate?: string | null
  actualEndDate?: string | null
  status?: RequirementStatusValue | null
}

/** 需求关联的测试计划（弹窗用） */
export interface RequirementPlanRef {
  planId: string
  planName: string
  releaseName?: string | null
  status: number
  lastRoundAt?: string | null
}
